components/calendar/src/calendar_arithmetic.rs

```diff
@@ -22,6 +22,30 @@ pub trait CalendarArithmetic: Calendar {
     fn month_days(year: i32, month: u8) -> u8;
     fn months_for_every_year(year: i32) -> u8;
     fn is_leap_year(year: i32) -> bool;
+
+    /// Calculate the days in a given year
+    /// Can be overridden with simpler implementations for solar calendars
+    /// (use `days_in_year_solar()`). Leave this as the default
+    /// for lunar calendars
+    ///
+    /// The name has `provided` in it to avoid clashes with Calendar
+    fn days_in_provided_year(year: i32) -> u32 {
+        let months_in_year = Self::months_for_every_year(year);
+        let mut days: u32 = 0;
+        for month in 1..=months_in_year {
+            days += Self::month_days(year, month) as u32;
+        }
+        days
+    }
+
+    /// Set days_in_year to this for solar calendars
+    fn days_in_year_solar(year: i32) -> u32 {
```

Comment:
Nit: I don't like having this method here since it should not be called as part of the trait; it's only a convenience. I'd rather see solar calendars implement days_in_provided_year by calling `Iso::days_in_provided_year()` and putting the custom impl on `Iso`.

Reply:
They can't call the ISO one because ISO has different leap year rules. I'll just copy paste the implementations instead.